core/libdivecomputer.c

```diff
@@ -113,14 +113,16 @@ static int parse_gasmixes(device_data_t *devdata, struct dive *dive, dc_parser_t
 			report_error("Warning: different number of gases (%d) and cylinders (%d)", ngases, ntanks);
 		} else if (ntanks > ngases) {
 			shown_warning = true;
-			report_error("Warning: smaller number of gases (%d) than cylinders (%d). Assuming air.", ngases, ntanks);
+			report_error("Warning: smaller number of gases (%d) than cylinders (%d).", ngases, ntanks);
 		}
 	}
 	bool no_volume = true;
+	cylinder_t cyl = empty_cylinder;
```

Comment:
Not sure why this is moved out of the loop - this seems to have an incorrect scope now, making the code harder to read / understand.

Reply:
Simply a minor optimization - no need to constantly wind/unwind stack frames.  Can revert. I like all variables to be defined (and initialized) at the top of a function but understand that most people prefer to place them throughout the code just prior to first use.

The loop contents should probably be a separate function anyway, overall the `parse_gasmixes` functions is -way- too long/complicated to easily understand. Parts of it perform data acquisition/parsing, parts of it perform the assignment of mixes to cylinders, etc.

Reply:
Not really an optimization and makes the code harder to reason about. The compiler should not have to rebuild the stackframe for every pass:
https://godbolt.org/z/q8PWzcE3Y
And if it does, I'd think that it is completely irrelevant performance-wise.

Reply:
Reverted.